### ISSUE TITLE
Leave-one-out rendering

### ISSUE THREAD

**Issue by grgkopanas** [intel-isl/StableViewSynthesis#8]
Hi,

I am trying to do a leave-one-out rendering for the input cameras, I have been doing it this way:
In this line https://github.com/intel-isl/StableViewSynthesis/blob/main/experiments/dataset.py#L263
I am adding a nbs.remove(idx) before passing it to the next function.

The results are rather extremely blurry, am I doing something wrong?

All the best,
George Kopanas

**Comment by griegler** [comment]
You should be able to just use/adapt [this method](https://github.com/intel-isl/StableViewSynthesis/blob/d6af53bcc8c30e6e5e8af2257e6deaf374413eb7/experiments/exp.py#L172). It creates an `Dataset` object with `mode=train`, which is exactly the leave-one-out behaviour that you want.

**Comment by griegler** [comment]
In more detail, add an entry [here](https://github.com/intel-isl/StableViewSynthesis/blob/d6af53bcc8c30e6e5e8af2257e6deaf374413eb7/experiments/exp.py#L318) with some name that calls a modified (such that it loads your models) [`get_train_set_tat`](https://github.com/intel-isl/StableViewSynthesis/blob/d6af53bcc8c30e6e5e8af2257e6deaf374413eb7/experiments/exp.py#L172) method. Then you can just use the provided code for evaluation and use your evaluation set name.

**Comment by grgkopanas** [comment]
Hi,

Thanks for the reply. I managed to use get_train_set_tat but I am quickly running out of GPU memory in train mode in a 16GB GPU. With get_eval_set_tat I can render just fine. 

Is there any quick and dirty way to use less GPU with get_train_set_tat ?

Cheers,
George

**Comment by griegler** [comment]
What do you use for `n_nbs`? You could try to reduce the number.

**Comment by grgkopanas** [comment]
I had that at default which seems to be 3. Seems already pretty small to me right?

**Comment by griegler** [comment]
Yes, that should be more than feasible. Batch size is set to 1, or?

**Comment by grgkopanas** [comment]
Batch size is 1, both for eval and train. I am attaching my git diff in case you notice something. Unfortunately I don't have a GPU with bigger memory available.

[svs_diff.txt](https://github.com/intel-isl/StableViewSynthesis/files/6300005/svs_diff.txt)
and my command line arguments are:
--net resunet3.16_penone.dirs.avg.seq+9+1+unet+5+2+16.single+mlpdir+mean+3+64+16 --cmd eval --iter last --eval-dsets tat-scene-hugo --eval-scale 0.45

Thanks again,
George

**Comment by akashsharma02** [comment]
> In more detail, add an entry [here](https://github.com/intel-isl/StableViewSynthesis/blob/d6af53bcc8c30e6e5e8af2257e6deaf374413eb7/experiments/exp.py#L318) with some name that calls a modified (such that it loads your models) [`get_train_set_tat`](https://github.com/intel-isl/StableViewSynthesis/blob/d6af53bcc8c30e6e5e8af2257e6deaf374413eb7/experiments/exp.py#L172) method. Then you can just use the provided code for evaluation and use your evaluation set name.

Hi @griegler Could you point to the code which removes the target image from the neighbors when the Dataset object is created with `mode=train`?

I was able to adapt my dataset and run evaluation, however I was wondering if there is actually leave one out rendering occuring? If the target image exists in the dataset, wouldn't the unprojected points on the mesh, always take points from the very same target image, or could you please clarify, if I understand something incorrectly. 

Perhaps, I need to define a `subseq` and explicitly add indices that are to be skipped in the source images, is that correct?
